app/search/searcher.py

```diff
@@ -0,0 +1,40 @@
+from text_query import TextQuery
+import external_search
+import full_text_search
+import indexing
+
+
+class Searcher(object):
+    """A utility class for searching repositories."""
+
+    def __init__(
+        self, repo, external_search_backends, enable_fulltext_search,
```

Comment:
Optional: Maybe we can deprecate and remove `external_search_backends`. Looks like we haven't used it for years. This command returns no values:

```
$ tools/console google.org/personfinder              
> config.get("external_search_backends")
> config.get_for_repo("japan", "external_search_backends")
```

We used that to provide better Japanese search quality in 2011 which wasn't possible with AppEngine back then. But we haven't use it after that.